web-registry/src/pages/MyEcocredits.tsx

```diff
@@ -72,12 +76,14 @@ const WrappedMyEcocredits: React.FC<WithBasketsProps> = ({ baskets }) => {
     baskets,
   );
   const basketsWithClasses = useBasketsWithClasses(baskets);
+  const { api } = useLedger();
 
   const [creditBaskets, setCreditBaskets] = useState<
     (QueryBasketResponse | undefined)[][]
   >([]);
   const [basketPutOpen, setBasketPutOpen] = useState<number>(-1);
   const [isProcessingModalOpen, setIsProcessingModalOpen] = useState(false);
+  const [selectedBasketDenom, setSelectedBasketDenom] = useState('');
```

Comment:
still in terms of consistency, what about just using `basketTakeOpen` (as for `basketPutOpen`)?
this works right now because we only have one type of action we can do on a basket (take) but in the future, if we want to add more CTAs and from the basket table, then we'll need to change that

Reply:
Now `setBasketTakeTokens` and moved logic out of modal so it can be in web-components